proto/proto_crypt.c

```diff
@@ -109,15 +110,19 @@ proto_crypt_secret(const char * filename)
 	if (f != stdin)
 		fclose(f);
 
-	/* Compute the final hash. */
+	/* Compute the final hash and wipe context state. */
 	SHA256_Final(K->K, &ctx);
 
 	/* Success! */
 	return (K);
 
 err2:
+	/* Close the file if it isn't stdin. */
 	if (f != stdin)
 		fclose(f);
+
+	/* Wipe context state. */
+	SHA256_Final(K->K, &ctx);
 err1:
 	free(K);
```

Comment:
Should this be `proto_crypto_secret_free(K)` here?  Because we've just written keying material into `K->K`.

Reply:
Yes, added as a REBASE commit.